### PR TITLE
libvirt: Assign fully-qualified domain names to master & bootstrap nodes

### PR DESCRIPTION
**Pull request by zeenix**
This is needed for gather logs from the nodes.

Related: https://github.com/openshift/release/pull/4395

### CODE REVIEW

**Comment by praveenkumar** [comment]
Tested and now we are able to gather the logs.

```
DEBUG Gather remote logs                           
DEBUG Collecting info from nested-9rmfc-master-0.nested.openshift.testing 
 EBUG Warning: Permanently added 'nested-9rmfc-master-0.nested.openshift.testing,192.168.126.11' (ECDSA) to the list of known hosts.
DEBUG Gathering master journals ...                
DEBUG Gathering master containers ...              
DEBUG time="2019-07-31T14:58:28Z" level=fatal msg="failed to connect: failed to connect: context deadline exceeded" 
DEBUG Waiting for logs ...                         
DEBUG Log bundle written to ~/log-bundle.tar.gz    
INFO Bootstrap gather logs captured here "/home/prkumar/clusters/nested/log-bundle-20190731145829.tar.gz" 
```

@abhinavdahiya this will allow us to collect the bootstrap logs in case of failure which will unblock us https://github.com/openshift/release/pull/4395 one. 

/approve

**Comment by zeenix** [comment]
@praveenkumar Cool. You also need to set the '/lgtm' label i think.

**Comment by praveenkumar** [comment]
> You also need to set the '/lgtm' label i think.

Yes, I just need @abhinavdahiya or @wking view once on that..

File: data/data/libvirt/bootstrap/main.tf [hunk 1]
```diff
@@ -34,7 +34,7 @@ resource "libvirt_domain" "bootstrap" {
 
   network_interface {
     network_id = var.network_id
-    hostname   = "${var.cluster_id}-bootstrap"
+    hostname   = "${var.cluster_id}-bootstrap.${var.cluster_domain}"
```
Comment (abhinavdahiya) [inline data/data/libvirt/bootstrap/main.tf]: does this affect the adoption by the machine-api?

Comment (abhinavdahiya) [reply data/data/libvirt/bootstrap/main.tf]: nvm this is bootstrap host.

File: data/data/libvirt/main.tf [hunk 2]
```diff
@@ -108,7 +109,7 @@ resource "libvirt_domain" "master" {
 
   network_interface {
     network_id = libvirt_network.net.id
-    hostname   = "${var.cluster_id}-master-${count.index}"
+    hostname   = "${var.cluster_id}-master-${count.index}.${var.cluster_domain}"
```
Comment (abhinavdahiya) [inline data/data/libvirt/main.tf]: does this affect the adoption by the machine-api?

Comment (abhinavdahiya) [reply data/data/libvirt/main.tf]: or how the node names are created?

Comment (zeenix) [reply data/data/libvirt/main.tf]: No, it doesn't but maybe we should do the same there for consistency.

Comment (zeenix) [reply data/data/libvirt/main.tf]: In any case, the provider change shouldn't be a blocker here.

Comment (zeenix) [reply data/data/libvirt/main.tf]: https://github.com/openshift/cluster-api-provider-libvirt/issues/166

**Pull request merged**